src/common/KokkosKernels_SparseUtils_mkl.hpp

```diff
@@ -0,0 +1,232 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef _KOKKOSKERNELS_SPARSEUTILS_MKL_HPP
+#define _KOKKOSKERNELS_SPARSEUTILS_MKL_HPP
+
+#include "KokkosKernels_config.h"
+
+#ifdef KOKKOSKERNELS_ENABLE_TPL_MKL
+
+#include <mkl.h>
+
+namespace KokkosSparse {
+namespace Impl {
+
+inline void mkl_internal_safe_call(sparse_status_t mkl_status, const char *name,
+                                   const char *file = nullptr,
+                                   const int line   = 0) {
+  if (SPARSE_STATUS_SUCCESS != mkl_status) {
```

Comment:
It would be nice to at least return the status value so the user gets an idea of what went wrong, even better would be a switch/if statement that assigns a string with error explanation.

Reply:
Fixed with https://github.com/kokkos/kokkos-kernels/pull/1365/commits/ea6da8d81de7a61fec7ada44ae411321f162560e on https://github.com/kokkos/kokkos-kernels/pull/1365